node/src/reactor/initializer.rs

```diff
@@ -164,6 +165,15 @@ impl Reactor {
         let contract_runtime =
             ContractRuntime::new(storage_config, &config.value().contract_runtime, registry)?;
 
+        if let Some(block) = storage.get_highest_block_header_for_db_check() {
+            let highest_block_header = block.take_header();
+            let blake_hash = Blake2bHash::new(highest_block_header.state_root_hash().as_ref());
+            let missing_trie_key = contract_runtime.retrieve_missing_keys(blake_hash);
+            if !missing_trie_key.is_empty() {
+                panic!("Fatal error! Trie-Key store is not empty. Wipe the DB to ensure operations")
+            }
+        }
```

Comment:
I'm wondering whether we should also validate the integrity of a DB even before having any blocks? If yes, then this can probably be part of the `CommitGenesis` process. Thoughts @xcthulhu ?

Reply:
> I'm wondering whether we should also validate the integrity of a DB even before having any blocks?

I think we need to do the following:

  1.  Check all of the blocks
  2. Check all of the block bodies
  3. Check all of the deploys/hashes
  4. Check all of the `state_root_hash`s for those blocks

I suspect that (4) will require we change `retrieve_missing_keys` to do a bulk check.